### PR TITLE
refactor: changed react components to use proper typescript types

### PR DESCRIPTION
**Pull request by nmerget**
## Proposed changes

- add typescript types for components to use them in react
- e.g. enables `DBInput` to use `aria-xxx` or `onMouseEnter` directly

## Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [x] Refactoring (fix on existing components or architectural decisions)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation Update (if none of the other choices apply)

### CODE REVIEW

File: packages/components/src/components/button/button.lite.tsx [hunk 1]
```diff
@@ -1,63 +1,13 @@
...
-			}
-		]
-	}
+	isAttachedToShadowDom: true
 });
 
 export default function DBButton(props: DBButtonProps) {
+	const ref = useRef<HTMLButtonElement>(null);
 	// This is used as forwardRef
 	let component: any;
```
Comment (mfranzke) [inline packages/components/src/components/button/button.lite.tsx:12]: @nmerget did you leave these in there intentionally ?

File: packages/components/src/shared/model.ts [hunk 2]
```diff
@@ -273,7 +272,6 @@ export type LinkProps = {
 		| undefined;
 	disabled?: boolean;
 	href?: string;
-	hreflang?: string;
```
Comment (mfranzke) [inline packages/components/src/shared/model.ts:296]: @nmerget what's the reason for this removal ?

Comment (nmerget) [reply packages/components/src/shared/model.ts:296]: TS said, it is not a valid attribute

Comment (nmerget) [reply packages/components/src/shared/model.ts:296]: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a?retiredLocale=de#hreflang

_No built-in functionality._ 

I don't know how to interpret this

File: showcases/react-showcase/src/components/form/index.tsx [hunk 3]
```diff
@@ -87,12 +90,13 @@ const FormComponent = () => {
 							type="date"
 						/>
 						<DBTextarea
+							aria-atomic
```
Comment (mfranzke) [inline showcases/react-showcase/src/components/form/index.tsx:93]: what's the reason to introduce this attribute?

File: packages/components/src/shared/model.ts (hunk 2 above)
Comment (mfranzke) [reply packages/components/src/shared/model.ts:296]: Me neither – at least it's included here: https://github.com/search?q=repo%3Amicrosoft%2FTypeScript%20hreflang&type=code

File: showcases/react-showcase/src/components/form/index.tsx (hunk 3 above)
Comment (nmerget) [reply showcases/react-showcase/src/components/form/index.tsx:93]: I tested if `aria-*` attributes are passed.

To be honest, `atomic` sounds cool, so I used this for the test :P

Comment (mfranzke) [reply showcases/react-showcase/src/components/form/index.tsx]: totally agreed – but let's use [`aria-label`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label) in this case, as it has more meaning and a clear use case for this element.

**Review by mfranzke — APPROVED** [review]

**Review by mfranzke — APPROVED** [review]

**Pull request merged**